src/Runner.Worker/FileCommandManager.cs

```diff
@@ -259,4 +259,99 @@ public void ProcessCommand(IExecutionContext context, string filePath, Container
+            if (ShouldUploadAttachment(context, filePath))
+            {
+                Trace.Info($"Submitting step summary content from file {filePath}, container: {container}");
+                var scrubbedFilePath = ScrubStepSummaryFileSecrets(filePath);
+
+                QueueStepSummaryUpload(context, scrubbedFilePath);
+            }
+        }
+
+        private bool ShouldUploadAttachment(IExecutionContext context, string filePath)
+        {
+            if (!ActionCommandManager.StepSummaryEnabled(context))
+            {
+                Trace.Info("Step Summary is disabled; skipping attachment upload");
+                return false;
+            }
+
+            if (String.IsNullOrEmpty(filePath))
+            {
+                Trace.Info("Step Summary path is empty; skipping attachment upload");
+                return false;
+            }
+
+            if (!File.Exists(filePath))
+            {
+                Trace.Info($"Step Summary file ({filePath}) does not exist; skipping attachment upload");
+                return false;
+            }
+
+            var fileSize = new FileInfo(filePath).Length;
+            if (fileSize == 0)
+            {
+                Trace.Info($"Step Summary file ({filePath}) is empty; skipping attachment upload");
+                return false;
+            }
+
+            if (fileSize > AttachmentSizeLimit)
+            {
+                context.Error($"$GITHUB_STEP_SUMMARY supports content up a size of {AttachmentSizeLimit/1024}k got {fileSize/1024}k");
+                Trace.Info($"Step Summary file ({filePath}) is too large ({fileSize} bytes); skipping attachment upload");
+                return false;
+
+            }
+
+            Trace.Info($"Step Summary file exists: {filePath} and has a file size of {fileSize} bytes");
+
+            return true;
+        }
+
+        private string ScrubStepSummaryFileSecrets(string filePath)
+        {
+            var scrubbedFilePath = filePath + "-scrubbed";
+
+            using (var streamReader = new StreamReader(filePath))
+            using (var streamWriter = new StreamWriter(scrubbedFilePath))
+            {
+                string line;
+                while ((line = streamReader.ReadLine()) != null)
+                {
+                    var maskedLine = HostContext.SecretMasker.MaskSecrets(line);
+                    streamWriter.WriteLine(maskedLine);
+                }
+            }
+
+            return scrubbedFilePath;
+        }
+
+        private void QueueStepSummaryUpload(IExecutionContext context, string filePath)
+        {
+            var stepID = context.Id;
+            var attachmentName = stepID.ToString();
+
+            Trace.Info($"Queueing file ({filePath}) for attachment upload ({attachmentName})");
+            try
+            {
+                context.QueueAttachFile(ChecksAttachmentType.StepSummary, attachmentName, filePath);
+            }
+            catch (Exception e)
+            {
```

Comment:
should customer see any warning or message if there step summary is not process properly.

Reply:
@TingluoHuang Good question! You mean like adding a warning visible to the customer when something went wrong? I think that's a great idea!

Reply:
warning or just info might want to double-check with PM.